graph/pull_v2.go

```diff
@@ -226,12 +240,28 @@ func (p *v2Puller) pullV2Tag(out io.Writer, tag, taggedName string) (verified bo
 			out.Write(p.sf.FormatProgress(stringid.TruncateID(img.ID), "Already exists", nil))
 			continue
 		}
+
+		digest := manifest.FSLayers[i].BlobSum
+		blobs := p.repo.Blobs(context.Background())
+		desc, err := blobs.Stat(context.Background(), digest)
+		if err != nil {
+			logrus.Debugf("Error statting layer: %v", err)
+			return false, err
+		}
+
+		totalSize += desc.Size
```

Comment:
Manifests sometimes reference the same layer twice. It would make this calculation more accurate to only count unique blobsums in the total.